hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -750,42 +781,42 @@ def _batch_spec(self):
         batch_spec = {'billing_project': self._client.billing_project, 'n_jobs': n_jobs, 'token': self.token}
         if self.attributes:
             batch_spec['attributes'] = self.attributes
-        if self.callback:
-            batch_spec['callback'] = self.callback
+        if self._callback:
+            batch_spec['callback'] = self._callback
         if self._cancel_after_n_failures is not None:
             batch_spec['cancel_after_n_failures'] = self._cancel_after_n_failures
         return batch_spec
 
-    async def _open_batch(self) -> Batch:
+    @unsubmitted_batch_only
+    async def _open_batch(self) -> int:
         batch_spec = self._batch_spec()
         batch_json = await (await self._client._post('/api/v1alpha/batches/create', json=batch_spec)).json()
-        self._update_id = batch_json['update_id']
-        return Batch(self._client,
-                     batch_json['id'],
-                     self.attributes,
-                     self.token,
-                     submission_info=BatchSubmissionInfo(False))
+        self._id = batch_json['id']
+        return int(batch_json['update_id'])
 
     def _update_spec(self) -> dict:
         update_token = secrets.token_urlsafe(32)
         return {'n_jobs': len(self._jobs), 'token': update_token}
 
-    async def _create_update(self, batch_id: int) -> int:
+    @submitted_batch_only
```

Comment:
If these are private methods, do they need these decorators? This feels like it would be dead code.